libraries/abstractions/common_io/include/iot_i2c.h

```diff
@@ -42,70 +42,69 @@
 
 /**
  * @brief I2C Bus status
  */
 typedef enum
 {
-    eI2CBusIdle = 0,
-    eI2cBusBusy = IOT_I2C_BUSY,
+    eI2CBusIdle = 0,            /*!< I2C bus is idle. */
+    eI2cBusBusy = IOT_I2C_BUSY, /*!< I2C bus is busy. */
 } IotI2CBusStatus_t;
 
 /**
  * @brief I2C operation status.
  */
 typedef enum
 {
-    eI2CCompleted = IOT_I2C_SUCCESS,
-    eI2CDriverFailed,
-    eI2CNackFromSlave = IOT_I2C_NACK,
-    eI2CMasterTimeout = IOT_I2C_BUS_TIMEOUT,
+    eI2CCompleted = IOT_I2C_SUCCESS,         /*!< I2C operation completed successfully. */
+    eI2CDriverFailed,                        /*!< I2C driver returns error during last operation. */
+    eI2CNackFromSlave = IOT_I2C_NACK,        /*!< Unexpected NACK is caught. */
+    eI2CMasterTimeout = IOT_I2C_BUS_TIMEOUT, /*!< I2C operation not completed within specified timeout. */
 } IotI2COperationStatus_t;
 
 /**
  * @brief I2C bus configuration
  */
 typedef struct IotI2CConfig
 {
-    uint32_t ulMasterTimeout;   /**<! Master timeout value in msec, to relinquish the bus if slave does not respond. This is an extended feature of SMBus which is not part of I2C protocol. */
-    uint32_t ulBusFreq;         /**<! Bus frequency/baud rate */
+    uint32_t ulMasterTimeout; /**<! Master timeout value in msec, to relinquish the bus if slave does not respond. This is an extended feature of SMBus which is not part of I2C protocol. */
+    uint32_t ulBusFreq;       /**<! Bus frequency/baud rate */
 } IotI2CConfig_t;
 
 /**
  * @brief Ioctl request types.
  */
 typedef enum
 {
-    eI2CSendNoStopFlag,     /**<! Set flag to not send stop after transaction */
-                            /**<! Default is always stop for every transaction */
-                            /**<! Flag will auto reset to stop after one transaction if you set no stop */
-    eI2CSetSlaveAddr,       /**<! All the operations use this slave address after it is set. */
-    eI2CSetMasterConfig,    /**<! Sets the I2C bus frequency and timeout using the struct IotI2CConfig_t,
-                             *   default speed is Standard */
-    eI2CGetMasterConfig,    /**<! Gets the I2C bus frequency and timeout set for the I2C master. */
-    eI2CGetBusState,        /**<! Get the current I2C bus status. Returns eI2CBusIdle or eI2CBusy */
-    eI2CBusReset,           /**<! Master resets the bus. */
-    eI2CGetTxNoOfbytes,     /**<! Get the number of bytes sent in write operation. */
-    eI2CGetRxNoOfbytes,     /**<! Get the number of bytes received in read operation. */
+    eI2CSendNoStopFlag,  /**<! Set flag to not send stop after transaction */
+                         /**<! Default is always stop for every transaction */
+                         /**<! Flag will auto reset to stop after one transaction if you set no stop */
+    eI2CSetSlaveAddr,    /**<! All the operations use this slave address after it is set. */
```

Comment:
Do we need to add anything about the expectations of the address?  7-bit address vs. 8-bit address vs. 10-bit address?

Reply:
Not sure what is 8-bit address. But I think both 7-bit and 10-bit should work as long as the driver supports.

Correspondingly, users need to give a uint8_t and uint16_t. Does that make sense to you?

I will clarify in next PR.